pkg/planner/core/index_join_path.go

```diff
@@ -596,12 +596,13 @@ func indexJoinPathRemoveUselessEQIn(buildTmp *indexJoinPathTmp, idxCols []*expre
 	return notKeyEqAndIn, nil
 }
 
-func getIndexJoinIntPKPathInfo(ds *logicalop.DataSource, innerJoinKeys, outerJoinKeys []*expression.Column) (
-	keyOff2IdxOff []int, newOuterJoinKeys []*expression.Column, ranges ranger.Ranges, ok bool) {
+func getIndexJoinIntPKPathInfo(ds *logicalop.DataSource, innerJoinKeys, outerJoinKeys []*expression.Column,
```

Comment:
just add a path as return value from this util function ,which will functioned more like: getBestIndexJoinPathResult, which the chosen path is integrated in indexJoinResult.